src/connection_manager.c

```diff
@@ -176,6 +176,46 @@ static void set_config_host(qd_server_config_t *config, qd_entity_t* entity)
     assert(config->host);
 }
 
+static char* qd_config_ssl_profile_get_password(qd_config_ssl_profile_t* ssl_profile)
+{
+    char *pw = ssl_profile->ssl_password;
+    if (!pw) return pw;
+
+    /* if the "password" starts with "env:" or "env: " then the remaining
+     * text is the environment variable that contains the password
+     */
+    if (strncmp(pw, "env:", 4) == 0) {
+        char *env = pw + 4;
+        /* skip the space if it is there */
+        if (*env == ' ') ++env;
+
+        const char* passwd = getenv(env);
+        if (passwd) {
+            free(ssl_profile->ssl_password);
+            pw = ssl_profile->ssl_password = strdup(passwd);
```

Comment:
Yes, this is replacing the value, and true, could cause problems if the literal value started with "env:".  Actually, the fix you mention would not be so great.  On the first time through the "literal" may or may not be handled depending on a subtle change in the code, and on the 3rd time, might have trouble or not.  I think that the "env:" and "literal:" keywords are supposed to be in-effect on the config file value, and should not apply to the env-var value.  That keeps it cleaner and simpler.  So, that would mean that I shouldn't re-write the ssl_profile value.  What do you think of just assigning the result of getenv() directly?  Can we change the ssl_password member to be a const char*?  Otherwise, I'd have to cast away const.